Adds a plastic recipe through chemistry

MylieDaniels opened the pull request:
<!-- The text between the arrows are comments - they won't be visible on your PR. -->
<!-- To label this PR, add the label(s) without the prefixes surrounded by brackets anywhere, for example: [LABEL] -->
<!-- PRs should at least have one area (A-) label and at least one category (C-) label -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Adds a chemical recipe for plastic sheets, which always requires at least 10 carbon, 20 hydrogen, and 10 oil to complete.
In addition, to stand in for a Phillips catalyst, we use chromium here. Without silicon dioxide, this chromium will deplete 40 units over the process.
Phillips catalysts sit on silica gel to prevent ashing, and to follow with this, silicon dioxide prevents the consumption of the catalyst by being consumed at a lower rate (2u per sheet)
If there's magnesium chloride present, which was discovered to increase polyethylene yields in the 1970s, it will prevent the carbon and hydrogen from consuming each other (you can also use fuel for this) and will speed the reaction up linearly over time - but 2u will be consumed per sheet.


## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Plastic is new, not powerful, and could probably use a source.
This is a way to introduce a material source that can be somewhat industrialized through prepwork.

## Changelog <!-- If necessary, put your changelog entry below. Otherwise, /please/ delete this entire section. -->
<!-- Put how you want to be credited in the changelog in place of CodeDude. -->
<!-- Use (*) for major changes and (+) for minor changes. See the contributor guide for details. For example: -->

```changelog
(u)MylieDaniels
(*)Plastic sheets can now be formed with 10 oil, 10 carbon, 20 hydrogen, and 40 chromium. Takes a while!
(+)Silicon dioxide will prevent chromium from depleting in plastic production, while magnesium dioxide will speed it up and prevent the carbon and hydrogen from fighting. Optimize!
```

TobleroneSwordfish reviewed (changes requested):
This is cool, I like recipes based on real-world processes, but it's a little overcomplicated I think for something that just produces a few low-value plastic sheets?
It also doesn't inhibit the carbon-hydrogen dissolving reaction so is almost impossible to complete right now.
Oh and test what happens when it occurs in smoke clouds.

MylieDaniels commented:
> This is cool, I like recipes based on real-world processes, but it's a little overcomplicated I think for something that just produces a few low-value plastic sheets?

I was a bit worried about making actual materials with chemistry as a concept, so started on the complex and technical side.
Do you have suggestions of what might make this more appealing or more fun to work with?

> It also doesn't inhibit the carbon-hydrogen dissolving reaction so is almost impossible to complete right now.

For making a low amount of plastic, I was producing it with fuel in the mix to make oil as it processed, and for bulk plasticmaking, this makes magnesium chloride inhibit the carbon-hydrogen reaction when in quantities greater than 5 units.

> Oh and test what happens when it occurs in smoke clouds.

Will do!

TobleroneSwordfish commented:
Oh I missed the oil recipe thing (sleepy brain), that's neat! For reducing complexity, I'd probably just remove the magnesium chloride interactions entirely, as they don't really add much other than making it (potentially infinitely?) faster and as it stands this is basically three different recipes combined into one. I'll see what other chemistry-interested devs think though.

MylieDaniels commented:
I've simplified that magnesium chloride logic and made it either 0.25 speed (40 second plastic) with, or 0.05 (200 second plastic) without. It will run at the same rate as unheated oil without, so the least industrial method is a room temperature large beaker of 10 carbon, 20 fuel, 20 hydrogen, and 40 chromium, which will produce a sheet of plastic in 200 seconds.

I've removed the ratio check for chromium, simplifying the recipe and allowing silicon dioxide to prop up the chromium content entirely.

The most industrial way, which requires more upfront prep but less time per sheet, would be to add a few units of chromium in a barrel, then add at least 2u silicon dioxide and 2u magnesium chloride per sheet you want to produce. You can then either produce the oil with hot fuel (it needs to be slightly heated to keep up with the faster pace) or add oil directly, but either way, 10 oil, 10 carbon, and 20 hydrogen will produce a sheet every 40 seconds.

Flaborized commented:
This is neat yeah, I like the idea of more physical, non-chemistry orientated products that you can make from chemistry. I would've also had the feedback about the complexity:usefulness ratio but the newer updated recipe seems more reasonable, good to have it be more complicated without actually being *that* difficult to make.

It could even be neat if we later expanded on this a bit and made plastic useful to research in some way ie: maybe letting you fabricate lids or something with it, but that's sorta beyond the scope here (maybe). Could do like anything, and would fill a neat niche.

Tarmunora commented:
I still think that using ethanol+sulfuric acid as the ethylene standin would be better but this is fine

TobleroneSwordfish commented on code/modules/chemistry/Chemistry-Recipes.dm:
```diff
@@ -2368,6 +2368,39 @@
+				holder.remove_reagent("silicon_dioxide", created_volume / 5)
+			else
+				holder.remove_reagent("chromium", 4 * created_volume) // super exaggerated catalytic ashing
+			if (holder.has_reagent("magnesium_chloride", 10))
```
```suggestion
			if (holder.has_reagent("magnesium_chloride"))
```
this is just going to confuse people otherwise
